packages/openapi3/package.json

```diff
@@ -44,15 +47,21 @@
     "test:ci": "vitest run --coverage --reporter=junit --reporter=default",
     "lint": "eslint . --max-warnings=0",
     "lint:fix": "eslint . --fix",
-    "regen-docs": "tspd doc .  --enable-experimental  --output-dir ../../docs/emitters/openapi3/reference"
+    "regen-docs": "tspd doc .  --enable-experimental  --output-dir ../../docs/emitters/openapi3/reference",
+    "generate-tsp": "node --no-warnings --experimental-specifier-resolution=node --loader ts-node/esm ./test/openapi-to-typespec/utils/generate-typespec.ts",
```

Comment:
we've been using `tsx` instead of ts-node which is always a pain to make work with esm to run typescript directly

Reply:
also is this just used as a test file right now?

Reply:
Oh awesome - I was running into issues with ts-node and 'borrowed' what the compiler package was doing - will look at tsx instead.

Yes - currently this is just to generate the snapshots. I used ts-node early on to make debugging sane.

Reply:
Removed this script - have a `regen-specs` script that just runs vitest with recording (like the samples package)

Reply:
just need to make sure this gets called in the CI, either via` pnpm test` or `pnpm regen-samples` (or both)

Reply:
Yes - I verified that `pnpm test` (and `pnpm test:ci`) does cause the specs to be regenerated and fails if there are any differences (`regen-samples` allows the differences through so that the author can commit/include them as part of the PR - just like for the samples package)